### DISCUSSION TITLE
List as input model with `embedded` Body does not work

### DISCUSSION THREAD

**Discussion by JoanFM** [fastapi/fastapi#11281] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import pytest
from fastapi import Body
from pydantic import BaseModel

@pytest.mark.asyncio
async def test_issue():

    class A(BaseModel):
        a: str

    app = FastAPI()

    @app.post("/func/")
    async def func_embedded(input: List[A] = Body(embed=True)) -> List[A]:
        return input

    async with AsyncClient(app=app, base_url="http://test") as ac:
        response2 = await ac.post("/func/", data={'input': []})

    assert response2.status_code == 200
```

### Description

When having a FastAPI with the Body(embed=True) and using a `List` the request fails with code 422

### Operating System

Linux

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

2.8.1

**Comment by YuriiMotov** [comment, marked as answer]
You should just use `json` parameter to pass JSON Body params:
```
        response2 = await ac.post("/func/", json={"input": []})
```

**Comment by JoanFM** [comment]
what is the reason why json works and not data in this case?
